docs/features/compilation-from-portable-pdb.md

```diff
@@ -0,0 +1,341 @@
+## compilation-from-portable-pdb
+
+Compilation from portable PDBs today is not completely possible, but is desirable in order to help reconstruct a compilation from source provided via source link or embedded in a pdb. Motivation is derived from [roslyn 41395](https://github.com/dotnet/roslyn/issues/41395).
+
+This document is restricted to the following assumptions:
+
+1. The benefit is for builds with `-deterministic` and published to the symbol server.
+2. Source generator and analyzer references are not needed for this task. They may be useful, but are out of scope for this document.
+3. Any storage capacity used for PDBs and source should not impact this feature, such as compression algorithm.
+4. Only Portable PDB files will be included for this spec. We can expand this feature past these once it is implemented and proven needed elsewhere.
+
+This document will provide the expanded specification to the Portable PDB format. Any additions to that format will be ported to expand documentation provided in [dotnet-runtime](https://github.com/jnm2/dotnet-runtime/blob/26efe3467741fe2a85780b2d2cd18875af6ebd98/docs/design/specs/PortablePdb-Metadata.md#source-link-c-and-vb-compilers).
+
+## List of Compiler Flags
+
+#### CSharp Flags That Can Be Derived From PDB or Assembly
+
+* baseaddress
+* checksumalgorithm
+* debug
+* deterministic
+* embed
+* filealign
+* highentropyva
+* link
+    - Represented by a metadata reference with `EmbededInteropTypes=true` 
+* linkresource
+    - Will be represented in metadata reference embedded in pdb
+* main
+    - Already stored in PDB as the entry point token
+* moduleassemblyname
+* modulename
+* nowin32manifest
+* pdb
+* platform
+* publicsign
+* resource
+    - Will be represented in metadata reference embedded in pdb
+* subsystemversion
+* target
+* win32icon
+* win32manifest
+* win32res
+
+#### CSharp Flags Not Included
+
+* appconfig
+* bugreport
+* delaysign
+* doc
+* errorendlocation
+* errorlog
+* errorreport
+* fullpaths
+* incremental
+* keycontainer
+* keyfile
+* noconfig
+* nologo
+* nostdlib
```

Comment:
This is an important option cause it impacts compilation. Can it be derived from the existing PDB info?

Reply:
nostdlib can be found because of metadata references, but I'm not sure how much it correlates directly. I meant to move this up and might have missed it

Reply:
I put in VB but not C# portion. Just needs to be moved

Reply:
`nostdlib` is only [used](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs#L1324) by the command line compiler when deciding what references to create for the compilation. We store a complete set of metadata references that were passed to the compilation in the PDB, so we don't need `nostdlib`.

Reply:
That's true. Essentially when re-creating the command line, should we choose to do that, we should always pass `/nostdlib` because what we store in the PDB is the final reference set.

Reply:
Yes. I believe it will be possible to create a command line, it won't obviously be the exactly same command line, but it should produce the same results (modulo signing and 
perhaps the PDB compression depending how deal with the compression lib). 

Note that the main goal is not really the command line, but rather a Compilation with all the options and parameters necessary. But as I said text version of the command line should also be possible.